src/components/calendarInput/index.jsx

```diff
@@ -0,0 +1,112 @@
+import CalendarIcon from 'assets/calendar.png';
+import { addDays, addYears, endOfYear, format, startOfMonth } from 'date-fns';
+import React, { useEffect, useRef, useState } from 'react';
+import { DayPicker } from 'react-day-picker';
+import { useDispatch, useSelector } from 'react-redux';
+import { setDate } from 'store/slices/inputsSlice';
+
+import { MainInput, MainInputImg, MainInputWrapper } from '../common/styled';
+import { DayPickerWrapper } from './styled';
+
+const DATE_FORMAT_PATTERN = 'iii d MMM';
+const today = new Date();
+
+export default function CalendarInput() {
+  const dates = useSelector(state => state.inputs.dates);
+  const [range, setRange] = useState(() => ({
+    from: dates.from ? new Date(dates.from) : null,
+    to: dates.to ? new Date(dates.to) : null,
+  }));
+  const [showCalendar, setShowCalendar] = useState(false);
+  const dispatch = useDispatch();
+
+  const notInitialRender = useRef(false);
+
+  useEffect(() => {
+    if (notInitialRender.current)
+      dispatch(
+        setDate({
+          from: range.from ? format(range.from, 'y-M-d') : null,
+          to: range.to ? format(range.to, 'y-M-d') : null,
+        }),
+      );
+    else notInitialRender.current = true;
+  }, [range]);
+
+  const handleRangeSelect = newRange => {
+    if (!newRange) {
+      setRange({ from: null, to: null });
+    } else if (!newRange.to) {
+      setRange({ ...newRange, to: null });
+    } else {
+      setRange(newRange);
+    }
+  };
+
+  let text = 'Check-in date -- Check-out date';
+  if (range?.from) {
+    if (!range.to) {
+      text = `${format(range.from, DATE_FORMAT_PATTERN)} -- Check-out date`;
+    } else if (range.to) {
+      text = `${format(range.from, DATE_FORMAT_PATTERN)} -- ${format(
+        range.to,
+        DATE_FORMAT_PATTERN,
+      )}
+      `;
+    }
+  }
```

Comment:
просто вынеси логику в функцию и вызывай её
```suggestion  
  function getInputText() {
    ...
  }
  
  const text = getInputText();
```

Reply:
вынес